concourse/client.py

```diff
@@ -188,12 +202,14 @@ class ConcourseApi(object):
 
     @param base_url: concourse endpoint (e.g. https://ci.concourse.ci)
     @param team_name: the team name used for authentication
+    @param concourse_version: major version of the target concourse
     @param verify_ssl: whether or not certificate validation is to be done
     '''
     @ensure_annotations
-    def __init__(self, base_url: str, team_name: str, verify_ssl=False):
+    def __init__(self, base_url: str, team_name: str, concourse_version: int, verify_ssl=False):
         self.base_url = base_url
         self.team = team_name
+        self.concourse_version = concourse_version
         self.routes = ConcourseApiRoutes(base_url=base_url, team=team_name)
```

Comment:
the api routes object should be passed-in by the factory function (in the proper version)